ourrecipes/src/components/pages/CreateRecipe.vue

```diff
@@ -0,0 +1,172 @@
+    <p>Recipe Name <input id="name" v-model="message" placeholder="Enter recipe name"></p>
+    <p>Recipe Description <input id="desc" v-model="message" placeholder="Enter recipe description"></p>
+    <p id="tagList">Tags 
+      
+      <input id="tag1" v-on:click="updateList" v-model="message" placeholder="Enter tag name">
+    </p>
+    <p id="ingredientList">Ingredients
+      
+      <input id="ingredient1" v-on:click="updateList" v-model="message" placeholder="Enter ingredient name">
+    </p>
+    <p id="instructionList">Instructions 
+      
+      <input id="instruction1" v-on:click="updateList" v-model="message" placeholder="Enter instruction name">
+    </p>
+    <button v-on:click="submitRecipe" class="createRecipe">Add Recipe</button>
+  </div>
+  <p id="submitting"></p>
+</template>
+
+<script>
+import db from '../firebaseInit';
+import firebase from '../firebaseApp'
+
+import RecipeDBO from '../dbCode';
+
+export default {
+  recipeName: "Recipe",
+  props: {
+  },
+  components: {
+    // NavBar,
+  },
+  mounted() {
+    let initTag = document.getElementById('tag1');
+    initTag.addEventListener('click', () => this.updateList(initTag))
+
+    let initInstruction = document.getElementById('instruction1');
+    initInstruction.addEventListener('click', () => this.updateList(initInstruction))
+
+    let initIngredient = document.getElementById('ingredient1');
+    initIngredient.addEventListener('click', () => this.updateList(initIngredient))
+
+    document.addEventListener('click', this.handleClickAwayFromInputList, true);
+
+    let recipeDB = new RecipeDBO();
+    recipeDB.testSet();
+
+  },
+  data(){
+    return{
+      author: '',
+      recipeId: 1,
+      instructions: [],
+      ingredients: [],
+      description: '',
+      tags: [],
+      tmpInp: null
+    }
+  },
+  methods: {
+    submitRecipe: function() {
+      const data = {
+        "name": document.getElementById("name").value,
+        "description": document.getElementById("desc").value,
+        "tags": [...document.getElementById('tagList').children].map(inp => inp.value),
+        "instructions": [...document.getElementById('instructionList').children].map(inp => inp.value),
+        "ingredients": [...document.getElementById('ingredientList').children].map(inp => inp.value),
+        "authorUID":firebase.auth().currentUser.email,
+        "created": (new Date()).toLocaleString()
+      }
+      let invalidInputs = []
+      for(let key in data) {
+        if(data[key] == '' || data[key] == []) {
+          invalidInputs.push(key)
+        }
+      }
+
+      if(invalidInputs.length > 0) {
+        alert('the following fields cannot be empty:\n  - ' + invalidInputs.join('\n  - '))
+      }
+      else {
+        let submitMsg = document.getElementById('submitting');
+        submitMsg.innerHTML = 'Adding Recipe...';
+
+        db.collection('recipes').add(data).then(() => {
+          submitMsg.innerHTML = 'Added!';
```

Comment:
this can be changes to .innerText, innerHTML is better to only use when you specifically need to put HTML inside another tag to avoid vulnerabilities.

Reply:
Yeah just change this and set this to be based off master instead of main and we're good to merge this in in my opinion, looks really good.